src/NBitcoin/ChainedHeader.cs

```diff
@@ -286,15 +286,19 @@ public ChainedHeader FindAncestorOrSelf(ChainedHeader chainedHeader)
         }
 
         /// <summary>
-        /// Finds the ancestor of this entry in the chain that matches the block hash given.
+        /// Finds the ancestor of this entry in the chain that matches the block hash, or optionally, the block height given.
         /// </summary>
         /// <param name="blockHash">The block hash to search for.</param>
+        /// <param name="height">Optional height to search for.</param>
         /// <returns>The ancestor of this chain that matches the block hash.</returns>
-        public ChainedHeader FindAncestorOrSelf(uint256 blockHash)
+        public ChainedHeader FindAncestorOrSelf(uint256 blockHash, int height = -1)
         {
             ChainedHeader currentBlock = this;
-            while ((currentBlock != null) && (currentBlock.HashBlock != blockHash))
+            while (currentBlock != null)
             {
+                if ((currentBlock.HashBlock == blockHash) || (currentBlock.Height == height))
```

Comment:
why not make it part of the while condition?

Reply:
Its easier to read like this

Reply:
but has more overhead. 

I guess if formatted correctly, having this in while statememnt will be readable

Reply:
I don't think there is more overhead.
I generally prefer readability because the CLI will anyway compile and optimize the code